src/server/initServer.ts

```diff
@@ -234,30 +239,40 @@ const initServer = (
   );
 
   /**
-   * Get all logs for a certain month
-   * @author Gabe Abrams
-   * @param {number} year the year to query (e.g. 2022)
-   * @param {number} month the month to query (e.g. 1 = January)
-   * @returns {Log[]} list of logs from the given month
-   */
+ * Get filtered logs based on provided filters
+ * @author Gabe Abrams, Yuen Ler Chow
+ * @returns {Log[]} list of logs that match the filters
+ */
   opts.app.get(
-    `${LOG_REVIEW_ROUTE_PATH_PREFIX}/years/:year/months/:month`,
+    `${LOG_REVIEW_ROUTE_PATH_PREFIX}`,
     genRouteHandler({
       paramTypes: {
-        year: ParamType.Int,
-        month: ParamType.Int,
         pageNumber: ParamType.Int,
+        filters: ParamType.JSON,
       },
       handler: async ({ params }) => {
-        // Get user info
+      // Get user info
         const {
-          year,
-          month,
           pageNumber,
           userId,
           isAdmin,
+          filters,
         } = params;
 
+        const {
+          dateFilterState,
+          contextFilterState,
+          tagFilterState,
+          actionErrorFilterState,
+          advancedFilterState,
+        } = filters as {
+          dateFilterState: DateFilterState,
+          contextFilterState: ContextFilterState,
+          tagFilterState: TagFilterState,
+          actionErrorFilterState: ActionErrorFilterState,
+          advancedFilterState: AdvancedFilterState,
```

Comment:
Can we make this into a type? `LogReviewerFilters` perhaps? That would be a folder containing an `index.ts` which was this thing:

```ts
dateFilterState: DateFilterState,
          contextFilterState: ContextFilterState,
          tagFilterState: TagFilterState,
          actionErrorFilterState: ActionErrorFilterState,
          advancedFilterState: AdvancedFilterState,
```

plus all the other types would be subtypes of this one and would be inside that folder too